src/layout/dom.ts

```diff
@@ -21,9 +24,24 @@ export function reset(): void {
     selectorMap = [];
     urlMap = {};
     idMap = new WeakMap();
+    regionMap = new WeakMap();
     if (Constant.DEVTOOLS_HOOK in window) { window[Constant.DEVTOOLS_HOOK] = { get, getNode, history }; }
 }
 
+export function regions(root: ParentNode): void {
+    for (let key in config.regions) {
+        if (config.regions[key] && "querySelectorAll" in root) {
```

Comment:
If this doesn't exist is there a way for us to know and check earlier rather than here? If it doesn't exist should we even run Clarity? My prefernce is that we check once at init time and not enable Clarity if its feature prereqs fail.

Reply:
This is mainly for nodes that can get mutation but do not support querySelectorAll. E.g. text nodes. We need to proactively look for regions as single page app modifies itself. In those cases we want to ensure that we ignore mutations on nodes like text, comment, etc. I will think through this a bit more.

Reply:
Please add above as comment. Not obvious that is why you need to check each case. 

---
In reply to: [395765133](https://github.com/microsoft/clarity-js/pull/170#discussion_r395765133) [](ancestors = 395765133)

Reply:
Done